django/forms/jinja2/django/forms/widgets/multiple_input.html

```diff
@@ -1,5 +1,4 @@
-{% set id = widget.attrs.id %}<ul{% if id %} id="{{ id }}"{% endif %}{% if widget.attrs.class %} class="{{ widget.attrs.class }}"{% endif %}>{% for group, options, index in widget.optgroups %}{% if group %}
-  <li>{{ group }}<ul{% if id %} id="{{ id }}_{{ index }}"{% endif %}>{% endif %}{% for widget in options %}
-    <li>{% include widget.template_name %}</li>{% endfor %}{% if group %}
-  </ul></li>{% endif %}{% endfor %}
-</ul>
+{% set id = widget.attrs.id %}<div{% if id %} id="{{ id }}"{% endif %}{% if widget.attrs.class %} class="{{ widget.attrs.class }}"{% endif %}>{% for group, options, index in widget.optgroups %}{% if group %}
+  <label>{{ group }}</label>{% endif %}{% for widget in options %}<div>
```

Comment:
I see we no longer have the grouping `ul` for each group. Wouldn’t we want to also have an extra `div` for that?

```diff
<div id="media">
  <div>
    <label for="media_0">
      <input type="radio" name="nestchoice" value="unknown" id="media_0" />
      Unknown
    </label>
  </div>
+ <div>
  <label id="media_1_label">Audio</label>
  <div>
    <label for="media_1_0">
      <input aria-labelledby="media_1_label media_1_0_label" type="radio" name="nestchoice" value="vinyl" id="media_1_0" />
      <span id="media_1_0_label">Vinyl</span>
    </label>
  </div>
+ </div>
</div>
```

Reply:
The `ul` was mainly there for style and to indent the inputs. I'm not sure we need anything for presentational reasons and part of me wonders if it is a bit inconsistent to only wrap the inputs which have a `group` (Note: I don't think we want to wrap inputs without a group name, as this would make standard un-nested inputs look rather odd). Anyhow, I've added a commit adding back these `divs` for now.